c/include/nvtx3/nvToolsExtCounters.h

```diff
@@ -34,46 +27,14 @@
 #define NVTX_EXT_COUNTERS_MODULEID 4
 #endif
 
+/* Indicates that the counters ID is not provided. */
+#define NVTX_COUNTERS_ID_NONE           0
 
-/** Identifier of the semantic extension for counters. */
-#define NVTX_SEMANTIC_ID_COUNTERS_V1 5
-
-/***  Flags to augment the counter value. ***/
-#define NVTX_COUNTERS_FLAG_NONE       0
-
-/**
- * Convert the fixed point value to a normalized floating point.
- * Use the sign/unsign from the underlying type this flag is applied to.
- * Unsigned [0f : 1f] or signed [-1f : 1f]
- */
-#define NVTX_COUNTERS_FLAG_NORM       (1 << 1)
-
-/**
- * Tools should apply scale and limits when graphing, ideally in a "soft" way to
- * to see when limits are exceeded.
- */
-#define NVTX_COUNTERS_FLAG_LIMIT_MIN  (1 << 2)
-#define NVTX_COUNTERS_FLAG_LIMIT_MAX  (1 << 3)
-#define NVTX_COUNTERS_FLAG_LIMITS \
-    (NVTX_COUNTERS_FLAG_LIMIT_MIN | NVTX_COUNTERS_FLAG_LIMIT_MAX)
+/** Static (user-provided, feed-forward) counter (group) IDs. */
+#define NVTX_COUNTERS_ID_STATIC_START  (1 << 24)
 
-/** Counter time scope **/
-#define NVTX_COUNTERS_FLAG_TIME_POINT       (1 << 5)
-#define NVTX_COUNTERS_FLAG_TIME_SINCE_LAST  (2 << 5)
-#define NVTX_COUNTERS_FLAG_TIME_UNTIL_NEXT  (3 << 5)
-#define NVTX_COUNTERS_FLAG_TIME_SINCE_START (4 << 5)
-
-/** Counter value type **/
-#define NVTX_COUNTERS_FLAG_VALUE_ABSOLUTE   (1 << 10)
-#define NVTX_COUNTERS_FLAG_VALUE_DELTA      (2 << 10) // delta to previous counter sample
-
-/** Counter visualization hints **/
-#define NVTX_COUNTERS_FLAG_INTERPOLATE      (1 << 14)
-
-/** Datatypes for limits union (value of `limitType`). */
-#define NVTX_COUNTERS_LIMIT_I64 0
-#define NVTX_COUNTERS_LIMIT_U64 1
-#define NVTX_COUNTERS_LIMIT_F64 2
+/** Dynamically (tool) generated counter (group) IDs */
+#define NVTX_COUNTERS_ID_DYNAMIC_START 4294967296  /* 1 << 32 */
```

Comment:
If this works as-is, great. I get the feeling that it may benefit from adding the literal suffix `ULL` (for `unsigned long long`) which is very likely to be a 64-bit type.

Reply:
We want to stay C89 compatible, and the "LL" suffix was a C99 addition.  The stdint.h types are C99 too, technically, but only very few compilers whine about that in -std=c89 mode, and only with `-Wall` or -`Wlong-long`.  According to the cppreference page for C integer literals, "The type of the integer constant is the first type in which the value can fit".  So this will be parsed as a signed 64-bit integer, which can then undergo implicit promotion to an unsigned 64-bit integer, regardless of whether "long" or "long long" is used for those types.  It would be assignable directly to int64_t and uint64_t.  I tried with a bunch of compilers on godbolt in C89 mode, and I could never get any of them to give me a warning about comparing a signed value with an unsigned value:  https://godbolt.org/z/8G1435rzd

I did however find that some 32-bit GCCs didn't work with this (`error: integer constant is too large for 'long' type`), but it works if the define is `((uint64_t)1 << 32)` instead of 4294967296.  Since all other compilers liked the expression (assuming they weren't whining about stdint.h), I think that reads a bit better and is closer to our existing pattern, so I think we should change to this.   Note the `32` will be an `int` literal, but the `<<` operator will promote it to the same type as the other argument -- in cases like this where I have two `int` literals and want to promote to 64-bit/float/double before doing the operator, I like to cast or use a suffix on only the first operand and let the other promote.

Reply:
I changed it to `((uint64_t)1 << 32)`, also in the payloads header.